corehq/apps/style/static/style/ko/components/inline_edit.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Component for an inline editing widget: a piece of text that, when clicked on, turns into a textarea.
+ *
+ * Parameters (all optional)
+ *  - value: Text to display and edit
+ *  - name: HTML name of textarea
+ *  - id: HTML id of textarea
+ *  - placeholder: Text to display when in read-only mode if value is blank
+ *  - inline: Whether or not to display widget in line with surrounding content. Defaults to false.
+ *  - rows: Number of rows in textarea.
+ *
+ * By default, the widget is client-side only, and it is up to the calling code to actually save the value
```

Comment:
@millerdev Good points. The main place we're planning to use this is app/form/module names and comments, and the spec also calls for moving them above the main accordions/tabs on those pages (mockup below)...which is going to mess up the current saving paradigm anyway, especially on module/form view (if you edit the name, but don't save the page, do we prevent you from switching tabs?). Hmm. Now I'm leaning towards always doing actual server-side saving.

<img width="625" alt="screen shot 2016-04-04 at 5 57 44 pm" src="https://cloud.githubusercontent.com/assets/1486591/14264957/c5cd0330-fa8e-11e5-89c7-25b8bfd27de5.png">